src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,242 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T element, Node<T> next, Node<T> prev) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            addNodeToEnd(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (size == 0) {
+            addFirstNode(value);
+            return;
+        }
+        if (index == 0) {
+            addToTop(value);
+            return;
+        }
+        if (index == size) {
+            addNodeToEnd(value);
+            return;
+        }
+        Node<T> currentNode;
+
+        if (index <= size / 2) {
+            int count = 0;
+            currentNode = first;
+            while (count != index) {
+                currentNode = currentNode.next;
+                count++;
+            }
+        } else {
+            int count = size - 1;
+            currentNode = last;
+            while (count != index) {
+                currentNode = currentNode.prev;
+                count--;
+            }
+        }
+        Node<T> newNode = new Node<>(value, currentNode, currentNode.prev);
+        currentNode.prev.next = newNode;
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
```

Comment:
The exception thrown should be `IndexOutOfBoundsException`, not `ArrayIndexOutOfBoundsException`, to maintain consistency across the methods.

Reply:
Done